docs/docs/usage/hooks.md

```diff
@@ -0,0 +1,249 @@
+The built-in commands are currently splitted in 3 groups:
+
+- the [initialization phase](#initialization)
+- the [dependencies management](#dependencies-managment).
+- the [publication phase](#publication).
+
+You will most probably need to perform some recurrent tasks between the installation and publication phases (housekeeping, linting, testing, ...)
+this is why PDM lets you define your own tasks/phases using [user scripts](#user-scripts).
+
+To provides full flexibility, PDM allows to [skip some hooks and tasks](#skipping) on demand.
+
+## Initialization
+
+The intialization phase should occurs only once in a project lifetime by running the [`pdm init`](cli_reference.md#exec-0--init) 
+command to initialize an existing project (prompt to fill the `pyproject.toml` file).
+
+They trigger the following hooks:
+
+- [`post_init`][pdm.signals.post_init]
+
+``` mermaid
+flowchart LR
+  subgraph pdm-init [pdm init]
+    direction LR
+    post-init{{Emit post_init}}
+    init --> post-init
+  end 
+```
+
+## Dependencies management
+
+The dependencies management is required for the developper to be able to work and perform the following:
+
+- `lock`: compute a lock file from the `pyproject.toml` requirements.
+- `sync`: synchronize (add/remove/update) PEP582 packages from the lock file and install the current project as editable.
+- `add`: add a dependency
+- `remove`: remove a dependency
+
+All thoses steps are directly available with the following commands:
+
+- [`pdm lock`](cli_reference.md#exec-0--lock): execute the `lock` task
+- [`pdm sync`](cli_reference.md#exec-0--sync): execute the `sync` task
+- [`pdm install`](cli_reference.md#exec-0--install): execute the `sync` task, preceded from `lock` if required
+- [`pdm add`](cli_reference.md#exec-0--add): add a dependency requirement, re-lock and then sync
+- [`pdm remove`](cli_reference.md#exec-0--remove): remove a dependency requirement, re-lock and then sync
+- [`pdm update`](cli_reference.md#exec-0--update): re-lock dependencies from their latest versions and then sync
+
+They trigger the following hooks:
+
+- [`pre_install`][pdm.signals.pre_install]
+- [`post_install`][pdm.signals.post_install]
+- [`pre_lock`][pdm.signals.pre_lock]
+- [`post_lock`][pdm.signals.post_lock]
+
+
+``` mermaid
+flowchart LR
+  subgraph pdm-install [pdm install]
+    direction LR
+
+    subgraph pdm-lock [pdm lock]
+      direction TB
+      pre-lock{{Emit pre_lock}}
+      post-lock{{Emit post_lock}}
+      pre-lock --> lock --> post-lock
+    end
+    
+    subgraph pdm-sync [pdm sync]
+      direction TB
+      pre-install{{Emit pre_install}}
+      post-install{{Emit post_install}}
+      pre-install --> sync --> post-install
+    end
+    
+    pdm-lock --> pdm-sync
+  end
+```
+
+## Publication
+
+As soon as you are ready to publish you package/app/libray, you will require the publication tasks:
+
+- `build`: build/compile assets requiring it
+- `package`: package you library/application into a Python package
```

Comment:
`package` may be a bit confusing here. In Python packaging world, `build` means the process the build system reads the configuration and produces sdist or wheel. It should cover the `package` phase. Also in PDM there are only `build` and `upload`, no such a phase named `package`.

Reply:
I merged it into the `build` description. I had been tricked by the `pdm.cli.publish.package` module